components/sup/src/http_gateway.rs

```diff
@@ -168,11 +168,25 @@ impl Server {
                 Err(_) => HTTP_THREAD_COUNT,
             };
 
-            let bind = server::new(move || {
+            let mut workers = server::new(move || {
                 let app_state = AppState::new(gateway_state.clone());
                 App::with_state(app_state).configure(routes)
-            }).workers(thread_count)
-            .bind(listen_addr.to_string());
+            }).workers(thread_count);
+
+            // On Windows the default actix signal handler will create a ctrl+c handler for the
+            // process which will disable default windows ctrl+c behavior and allow us to
+            // handle via check_for_signal in the supervisor service loop. However, if the 
+            // supervsor is in a long running non-run hook, that loop will not get to 
+            // check_for_signal in a reasonable amount of time and the supervisor will not 
+            // respond to ctrl+c. On Windows, we let the launcher catch ctrl+c and gracefully
+            // shut down services. ctrl+c should simply halt the supervisor. The IgnoreSignals
+            // feature is always enabled in the Habitat Windows Service which relies on ctrl+c
+            // signals to stop the supervisor.
+            if feat::is_enabled(feat::IgnoreSignals) {
+                workers = workers.disable_signals();
+            }
```

Comment:
Is there ever a time where we would _not_ want to disable this signal handling behavior on Windows?

I'm just wondering if this kind of logic might better live in an `if cfg!(windows) {...}` block, instead.

Also, is this kind of conditional logic necessary only because long-running hooks can still block the main Supervisor thread? If so, we should file an issue to remove it once all the hooks are running on their own threads.

Reply:
Yes. Ideally we would allow ourselves to handle the signals. That would allow us to more safely clean up after ourselves. And under many scenarios, handling the signals ourdelves works well. It mainly comes down to the long running `init` hook scenarios where things can go wrong. I will file an issue to address their removal on the happy days when we handle the other hooks in their own threads.